keps/sig-testing/3041-node-conformance-and-features/README.md

```diff
@@ -0,0 +1,782 @@
+label, as `Feature` label indicates that the special environment configuration
+is needed to run tests in the “standard” CI.
+See the previous section and definition [here](https://github.com/kubernetes/community/blob/32a1c14d04ff78684d78b827ac7c49f70352d509/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests).
+
+The use of these labels in tests today is indicative of the labels' different
+meaning. Looking at tests configuration, wildcard `[NodeFeature:*`
+label is always used as focus and individual `NodeFeatures` are used in skip.
+It is opposite for the `Feature` label. The whildcard `[Feature:*` is always
+used to skip tests, while individual tests are present in focus.
+
+The reason for this difference is that runtimes that are being tested support
+all the NodeFeatures today and there is no need to list all NodeFeatures
+individually in test definitions. If we will have more fragmented support of
+`NodeFeatures` in runtimes being tested in CI, labels will have exactly the
+same semantics.
+
+Also both labels, `NodeFeature` and `Feature` degraded over time - they weren’t
+applied using this semantics.
+
+This KEP proposes to adjust the definition of the `Feature:` label which makes
+definitions compatible. The proposal is to unify `NodeFeatures` and `Features`
+and start relying on alternative labels for filtering in environments.
+
+- Rename NodeFeature to Feature. https://testgrid.k8s.io/sig-node-containerd#node-e2e-features will execute tests for all Features, excluding everything that needs a special environment, see labels above.
+- Document the meaning of NodeConformance. NodeConformance will indicate ALL tests that are testing enabled out of the box functionality that is not runtime or k8s distro specific. These tests may still require an additional environment set up. NodeConformance tests may have all labels specifying its environment requirements, like Slow, Disruptive, Special, etc.
+- Introduce a new periodic job that runs all NodeConformance tests with all FeatureGates turned off.
+
+### NodeConformance
+
+NodeConformance label represents tests that have to be working on all environments
+and k8s distros. The name is similar to the `[Conformance]` cluster tests that
+are part of the Kubernetes Conformance Tests, which serves as the base for the
+Kubernetes certification program. The similarity in the name may be confusing,
+but this confusion will not be addressed in this KEP.
+
+One difference proposed in this document is to allow NodeConformance to be
+applied to `[Beta]` Features. Since beta features are enabled out of the box and
+in most deployments all Beta features stay enabled, applying Feature label to
+these tests may be misleading. This difference is intended, as we want to make
+sure Beta features as enabled out of the box are tested on PR validation and
+across different environments. This will give a better signal for GA-ing the
+feature.
```

Comment:
As long as tests are consistently tagged `[Beta]`, then I don't see a problem with this, but new Beta's are off-by-default https://github.com/kubernetes/enhancements/issues/3136 so this might get a bit tricky.

Reply:
APIs are off, the features are on, meanwhile the feature doesn't use a v1beta1 ... API endpoint, it is enabled. I think that for node most feature doesn't depend on API endpoints, right?
anyway, this should be explained here as it is misleading